### PR TITLE
Update composer.json

### PR DESCRIPTION
**Pull request by akaneru**
Requirements updates in order to use the package with the version 3 of the framework.

### CODE REVIEW

**Comment by akaneru** [comment]
By accepting that PR will be able to work with this package also in ZF3.
Thanks!

**Comment by akaneru** [comment]
Please, anyone can handle that PR?
Thanks!

**Comment by akaneru** [comment]
Hi @mwillbanks, could you kindly handle that pull request?
Thanks

File: composer.json [hunk 1]
```diff
@@ -21,6 +21,12 @@
             "ZendServiceTest\\Google\\": "tests/"
         }
     },
+    "repositories": [
```
Comment (froschdesign) [inline composer.json]: What are the reasons for this extension?

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,7 @@
 language: php
 
+dist: trusty
```
Comment (froschdesign) [inline .travis.yml]: The same here.

File: composer.json (hunk 1 above)
Comment (akaneru) [inline composer.json]: I've done this relying on this https://github.com/zendframework/ZendService_Apple_Apns/blob/master/composer.json

File: .travis.yml (hunk 2 above)
Comment (akaneru) [inline .travis.yml]: Travis CI failed without this...

File: composer.json (hunk 1 above)
Comment (froschdesign) [inline composer.json]: Please look at this template for Composer. (still work in progress, but already the right boilerplate)

File: .travis.yml (hunk 2 above)
Comment (froschdesign) [inline .travis.yml]: Please ignore HHVM. You will find some background in this blog post: "[Zend Framework and PHP 7.1](https://framework.zend.com/blog/2017-06-06-zf-php-7-1.html)"

For Travis is also a template in work: is also a template in work: zendframework/maintainers#16

**Comment by akaneru** [comment]
Hi @froschdesign, with the new composer.json / travis file we have new errors:
`iconv_set_encoding(): Use of iconv.internal_encoding is deprecated`
That's strange because it appears to be already resolved https://github.com/zendframework/zf1/issues/344.
But maybe there's something that I'm missing or I don't know :|
Could you give me some suggestion?

Thank you!

**Comment by akaneru** [comment]
Hi @froschdesign, please let me know what can I do to solve the issue.
Best regards,
tomaso

**Comment by akaneru** [comment]
Hi again, an other updates about the Travis CI fail: locally I could run tests with success (below the result).

vendor/bin/phpunit --color=always
PHPUnit 4.8.36 by Sebastian Bergmann and contributors.

.............................S..

Time: 46 ms, Memory: 6.00MB

OK, but incomplete, skipped, or risky tests!
Tests: 32, Assertions: 108, Skipped: 1.
`

Thank you

**Comment by akaneru** [comment]
Hello, I've found a way to solve the issue by removing the package from requirements, add adding it to the autoload section in composer json file.
That allow me to update zend framework 3 and use it with the Gcm service.

    {
        "require": {
            "php": "^5.6|^7.0",
            "zendframework/zendframework": "^3.0",
            "zendframework/zend-i18n": "^2.7",
            "zendframework/zend-mvc-plugin-flashmessenger": "^1.0",
            "zendframework/zendservice-apple-apns": "dev-master"
        },
        "autoload": {
            "psr-4": {
                "ZendService\\Google\\": "libraries/ZendService_Google_Gcm/library/"
            }
        }
    }

Best regards.

**Pull request closed without merging**